WordPress/gutenberg
# Refactor `QueryControls` component to TypeScript

## What?
Convert the `QueryControls` component to TypeScript

## Why?
As part of an [effort](https://github.com/WordPress/gutenberg/issues/35744) to convert `@wordpress/components` to TypeScript

## How?
Mainly by adding types to the `QueryControls` component

## Testing Instructions
1. Create a new post
2. Add a 'Latest Posts' block
3. Expected: The 'Sorting and filtering' panel works the same as before:
<img width="846" alt="Screen Shot 2022-12-21 at 6 09 22 PM" src="https://user-images.githubusercontent.com/4063887/209027062-0c6b5051-e1df-44d4-9368-12a809244006.png">
(That uses this QueryControls component)

## Screenshots

See above
